src/Vault.sol

```diff
@@ -528,7 +545,17 @@ contract Vault is IERC4626, ERC20Permit, ILiquidationSource, IClaimable, Ownable
     bytes32 _r,
     bytes32 _s
   ) external returns (uint256) {
-    _permit(IERC20Permit(address(_asset)), _owner, address(this), _assets, _deadline, _v, _r, _s);
+    if (_owner != msg.sender) {
+      revert PermitCallerNotOwner(msg.sender, _owner);
+    }
+
+    IERC20Permit(address(_asset)).permit(_owner, address(this), _assets, _deadline, _v, _r, _s);
+
+    uint256 _allowance = _asset.allowance(_owner, address(this));
```

Comment:
I just realized that this conditional isn't needed anymore becaause of the `msg.sender` check

If the user doesn't have the allowance, then the `safeTransferFrom` in the `_depositAssets` will simply fail.

What we are protecting from is anyone being able to deposit on anyone's behalf if that person has set an allowance

Reply:
It's true we don't need it to solve the issue of depositing on someone else's behalf, but if we don't have the check, it can break the expectations of the caller. The example I mentioned before demonstrates the issue:

1. Alice manually approves `unlimited` WETH on the vault
2. Alice uses `depositAndPermit` with an updated limit of 100 WETH (what she's depositing)
3. After the deposit, Alice would expect that the new approval is at `0` since she decreased it when she deposited, but it's actually still `unlimited` since the permit didn't change anything.

Reply:
I see; that make sense.